editor.planx.uk/src/lib/gis.ts

```diff
@@ -0,0 +1,41 @@
+import { Feature } from "geojson";
+import { Dispatch, SetStateAction, useEffect, useState } from "react";
+
+type Projection = "EPSG:3857" | "EPSG:27700";
+
+export type GeoJSONChange = Record<Projection, { features: Feature[] }>;
+export type GeoJSONChangeEvent = CustomEvent<GeoJSONChange>;
+
+const isGeoJSONChangeEvent = (event: Event): event is GeoJSONChangeEvent => {
+  return event instanceof CustomEvent;
+};
+
+type UseGeoJSONChange = (
+  callback: (event: GeoJSONChangeEvent) => void,
+) => [Feature[] | undefined, Dispatch<SetStateAction<Feature[] | undefined>>];
+
+/**
+ * Hook for interacting with @opensystemslab/map
+ * Assign a callback function to be triggered on the "geojsonChange" event
+ */
+export const useGeoJSONChange: UseGeoJSONChange = (callback) => {
```

Comment:
I initially move this logic out of the components, and into the provider. 

I then realised it's actually possibly fairly generic / reusable and not MapAndLabel specific at all.

Not totally sold on location within the codebase - maybe we need a helpers folder for the map component or something?

Reply:
This is a nice idea! 

One nit on first read through: I think we need the `id` on L32 below to be a param as this vary throughout our various implementations (map&label, drawboundary, etc)!

Reply:
Ah yep, very good catch again thank you! Fixed in [f5bb2f4](https://github.com/theopensystemslab/planx-new/pull/3595/commits/f5bb2f4e84670f032f2472ff2d765994710038ee)